src/Resources/Tags/Tag/NewAzureTagCommand.cs

```diff
@@ -12,30 +12,84 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using System.Management.Automation;
 using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.Tags.Client;
 using Microsoft.Azure.Commands.Tags.Model;
 
 namespace Microsoft.Azure.Commands.Tags.Tag
 {
     /// <summary>
-    /// Creates a new tag with the specified values
+    /// Creates a predefined Azure tag or adds values to an existing tag | Creates or updates the entire set of tags on a resource or subscription.
     /// </summary>
-    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Tag"), OutputType(typeof(PSTag))]
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Tag", SupportsShouldProcess = true), OutputType(typeof(PSTag), typeof(PSTagResource))]
     public class NewAzureTagCommand : TagBaseCmdlet
     {
-        [Parameter(Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Name of the tag. If the tag name doesn't exist, create the tag name. Otherwise, add the value to the existing tag name.")]
+        #region Parameter Set Names
+
+        private const string CreatePredefinedTagSet = "CreatePredefinedTagSet";
+        private const string CreateByResourceIdParameterSet = "CreateByResourceIdParameterSet";
+
+        #endregion
+
+        #region Input Parameter Definitions
+
+        [Parameter(Position = 0, 
+            Mandatory = true, 
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CreatePredefinedTagSet,
+            HelpMessage = "Name of the tag. If the tag name doesn't exist, create the tag name. Otherwise, add the value to the existing tag name.")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(Position = 1, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Value of the tag. If specified, add the tag value to the tag name. Otherwise, keep the tag value unchanged.")]
+        [Parameter(Position = 1, 
+            Mandatory = false, 
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CreatePredefinedTagSet,
+            HelpMessage = "Value of the tag. If specified, add the tag value to the tag name. Otherwise, keep the tag value unchanged.")]
         [ValidateNotNullOrEmpty]
         public string Value { get; set; }
 
+        [Parameter(Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CreateByResourceIdParameterSet,
+            HelpMessage = "Creates or updates the entire set of tags on resourceId, resourceId can be a resource or subscription.")]
```

Comment:
The resource identifier for the entity being tagged. A resource, a resource group or a subscription may be tagged.